go-controller/pkg/util/util.go

```diff
@@ -139,3 +139,36 @@ func IsAnnotationNotSetError(err error) bool {
 	_, ok := err.(annotationNotSetError)
 	return ok
 }
+
+// CalculateHostSubnetsForClusterEntry calculates the host subnets
+// available in a CIDR entry
+func CalculateHostSubnetsForClusterEntry(cidrEntry config.CIDRNetworkEntry,
+	v4HostSubnetCount, v6HostSubnetCount *int) {
+	prefixLength, _ := cidrEntry.CIDR.Mask.Size()
+	if prefixLength > cidrEntry.HostSubnetLength {
+		klog.Warningf("Invalid cidr entry: %+v found while calculating subnet count",
+			cidrEntry)
+		return
+	}
+	if !utilnet.IsIPv6CIDR(cidrEntry.CIDR) {
+		*v4HostSubnetCount = *v4HostSubnetCount + 1<<(cidrEntry.HostSubnetLength-prefixLength)
+	} else {
+		*v6HostSubnetCount = *v6HostSubnetCount + 1<<(cidrEntry.HostSubnetLength-prefixLength)
```

Comment:
this can overflow :thinking:

Reply:
Yes, you are right. Updated the counters to float64.